File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/front/top/TopView.java
package com.ikmr.banbara23.yaeyama_liner_checker.front.top;

import com.ikmr.banbara23.yaeyama_liner_checker.front.base.BaseView;
import com.ikmr.banbara23.yaeyama_liner_checker.model.Company;

/**
 * 港詳細のViewインターフェイス
 */
public interface TopView extends BaseView {

    // 安栄・八重山観光フェリー・ドリーム
    void onClickedCompany(Company company);

    // 時刻表
    void onClickedTimeTable();

    // 設定
    void onClickedSetting();
}

File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/front/top/TopView.kt
package com.ikmr.banbara23.yaeyama_liner_checker.front.top

import android.content.Context
import com.ikmr.banbara23.yaeyama_liner_checker.front.base.BaseView
import com.ikmr.banbara23.yaeyama_liner_checker.model.Company

/**
 * 港詳細のViewインターフェイス
 */
interface TopView : BaseView {

    fun getContext(): Context

    fun navigateToWeather()

    fun navigateToTyphoon()

    fun navigateToCompanyStatusList(company: Company)

    fun navigateToPortStatusList(portName: String, portCode: String)
}
